bindings/python/render.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+
+import mako.template
+
+
+class ScannerSpec(object):
+    def __init__(self, state_t="size_t", extra_methods=(), ignored_methods=()):
+        self.state_t = state_t
+        self.extra_methods = extra_methods
+        self.ignored_methods = ignored_methods
+
+
+class OptionSpec(object):
+    def __init__(self, cpp_getter, letter=""):
+        assert len(letter) <= 1
+        self.cpp_getter = cpp_getter
+        self.letter = letter
+
+
+MAKO_GLOBALS = {
+    "FSM_BINARIES": [
+        ("+", "add", "const Fsm&", "Fsm"),
+        ("|", "or", "const Fsm&", "Fsm"),
+        ("&", "and", "const Fsm&", "Fsm"),
+        ("*", "mul", "size_t", "size_t"),
+    ],
+    "FSM_INPLACE_UNARIES": [
+        "AppendDot",
+        "Surround",
+        "Iterate",
+        "Complement",
+        "MakePrefix",
+        "MakeSuffix",
+        "PrependAnything",
+        "AppendAnything",
+        "Reverse",
+        "Canonize",
+        "Minimize",
+    ],
+    "OPTIONS": {
+        "LATIN1": OptionSpec("Pire::Encodings::Latin1()", "l"),
+        "UTF8": OptionSpec("Pire::Encodings::Utf8()", "u"),
+        "I": OptionSpec("Pire::Features::CaseInsensitive()", "i"),
+        "ANDNOT": OptionSpec("Pire::Features::AndNotSupport()", "a"),
+        "GLUE_SIMILAR_GLYPHS": OptionSpec("Pire::Features::GlueSimilarGlyphs()", "g"),
+    },
+    "SCANNERS": {
+        "Scanner": ScannerSpec(),
+        "NonrelocScanner": ScannerSpec(),
+        "ScannerNoMask": ScannerSpec(),
+        "NonrelocScannerNoMask": ScannerSpec(),
+        "SimpleScanner": ScannerSpec(ignored_methods={"AcceptedRegexps", "Glue"}),
+        "SlowScanner": ScannerSpec(
+            state_t="yvector[size_t]",
+            ignored_methods={"Glue", "Size", "LettersCount"}
+        ),
+        "CapturingScanner": ScannerSpec(
+            state_t="__nontrivial__",
+            ignored_methods={"AcceptedRegexps", "Glue"},
+        ),
+        "CountingScanner": ScannerSpec(
+            state_t="__nontrivial__",
+            ignored_methods={"AcceptedRegexps"},
+        ),
+    },
+    "SPECIAL_CHARS": [
+        "Epsilon",
+        "BeginMark",
+        "EndMark",
+        "MaxCharUnaligned",
+        "MaxChar",
+    ],
+}
+
+
+def make_argparser():
+    parser = argparse.ArgumentParser()
```

Comment:
I added the description, but the formatter you suggest prints the following:
```
  -o OUTPUT, --output OUTPUT
                        Path to place the rendered source at (default: None)
```
IMHO this `default: None` is misleading, as `output` is a required parameter.